### PR TITLE
Registry CLI implemented. Searching for protocols and skills added.

### PR DESCRIPTION
**Pull request by panasevychol**
## Proposed changes

Registry CLI implemented. Searching for protocols and skills added.
Not yet integrated with main CLI.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../master/CONTRIBUTING.rst) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

DELETE INCLUSIVE THIS AND BELOW FOR STANDARD PR
------

## Release summary

Version number: [e.g. 1.0.1]

## Release details

Describe in short the main changes with the new release.

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../master/CONTRIBUTING.rst) doc
- [ ] I am making a pull request against the `master` branch (left side), from `develop`
- [ ] Lint and unit tests pass locally
- [ ] I built the documentation and updated it with the latest changes
- [ ] I've added an item in `HISTORY.rst` for this release
- [ ] I bumped the version number in the `aea/__version__.py` file.
- [ ] I bumped the version number in every Docker image of the repo and published it. Also, I built and published them with tag `latest`  
      (check the READMEs of [`aea-develop`](../develop-image/README.md#publish) 
      and [`aea-deploy`](../deploy-image/README.md#publish))
- [ ] I have checked that the documentation about the `aea cli` tool works

## Further comments

Write here any other comment about the release, if any.

### CODE REVIEW

File: aea/cli/registry/main.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import click
```
Comment (DavidMinarsch) [inline aea/cli/registry/main.py]: This `main.py` is not required. We already instantiate a cli here: https://github.com/fetchai/agents-aea/blob/develop/aea/cli/__main__.py

File: aea/cli/registry/search.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import click
+
+from utils import format_list, request_api
+
+
+@click.group()
+def search():
```
Comment (DavidMinarsch) [inline aea/cli/registry/search.py]: Note, we have a cli search command implemented here: https://github.com/fetchai/agents-aea/blob/develop/aea/cli/search.py

File: aea/cli/registry/utils.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import click
```
Comment (DavidMinarsch) [inline aea/cli/registry/utils.py]: This could be placed in `aea/cli/registry_api.py` for instance

File: aea/cli/registry/utils.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import requests
+
+from settings import API_URL
```
Comment (DavidMinarsch) [inline aea/cli/registry/utils.py]: API_URL replaced `DEFAULT_REGISTRY_PATH`

File: aea/cli/registry/utils.py (hunk 3 above)
Comment (panasevychol) [reply aea/cli/registry/utils.py]: I think it's better to keep this methods isolated for now. Not to create a mess in old CLI module.

File: aea/cli/registry/utils.py (hunk 4 above)
Comment (panasevychol) [reply aea/cli/registry/utils.py]: `DEFAULT_REGISTRY_PATH ` leads us to a folder path when `API_URL ` is a web path. A different logic uses those constants (local dir search and search request to API).

**Pull request merged**